src/app/application.cpp

```diff
@@ -173,6 +174,12 @@ Application::Application(int &argc, char **argv)
     SettingsStorage::initInstance();
     Preferences::initInstance();
 
+    const bool firstTimeUser = !Preferences::instance()->getAcceptedLegal();
+    if (!firstTimeUser) {
+        if (!upgrade())
+            throw RuntimeError(u"Failed migration of old settings"_qs); // Not translatable. Translation isn't configured yet.
+    }
```

Comment:
The first change is here.

Reply:
This is the only change.
I initially had moved `handleChangedDefaults()` back to main.cpp
It runs very early here before the first time user is asked to accept. I was worried that it would unnecessarily create a settings file here and later the user declines when asked.
However, `handleChangedDefaults()` checks if the key exists in the file before writing values. So it should be safe to call it here.